### DISCUSSION TITLE
Tree-shakable third-party (user's) components like with Vuetify loader

### DISCUSSION THREAD

**Discussion by iliubinskii** [quasarframework/quasar#12666] [Ideas / Proposals]
Is it possible to add something like this:
https://vuetifyjs.com/en/features/treeshaking/#custom-dynamic-imports
to quasar?

**Comment by MilosPaunovic** [comment]
Quasar is already, by default, doing that for you.
[https://quasar.dev/start/upgrade-guide](https://quasar.dev/start/upgrade-guide#:~:text=Remove%20quasar.conf.js%20%3E%20framework%20%3E%20importStrategy.%20Auto%20import%20works%20so%20great%20that%20is%20now%20used%20by%20default%20and%20as%20the%20only%20option.)

**Comment by iliubinskii** [comment]
Do you want to say that quasar is tree shaking third-party components added with app.use():
```ts
export default boot(({ app }) => {
  app.use(plugin);
```
?

I think it is tree shaking ONLY its own components.

Vuetify allows to also tree-shake third party components and they use webpack plugin for that:
![image](https://user-images.githubusercontent.com/70687432/156887931-71a0bd28-0ac3-477f-ba6c-5cc551b18079.png)

**Reply by MilosPaunovic** [reply]
Yeah, Quasar is treeshaking its own components.
